File: 백준/Silver/14916. 거스름돈/거스름돈.py
n = int(input())
dp=[-1]*(n+10)
dp[2]=1
dp[5]=1
#작은 수부터 조건문 봐야함
#안그러면 4같은 친구들은 이미 -1로 초기화 되어 있는데
# 2라는 값을  min하면 갱신이 안됨
for i in range(3,n+1):
    if dp[i-2]!=-1:
        dp[i]=dp[i-2]+1
    if dp[i-5]!=-1 and i>5:
        dp[i]=min(dp[i-5]+1,dp[i])

print(dp[n])

File: 백준/Silver/14916. 거스름돈/거스름돈.cc
#include <iostream>
#include <string>
#include<math.h>
#include<cstring>
#include<climits>
#include<vector>
using namespace std;

int dp[100010];

int main() {


    int n;
    cin >> n;
  
    dp[3] = 1;
    dp[5] = 1;
    for (int i = 1; i <= 100004; i++)
    {
        dp[i] = 987654321;
    }
    dp[2] = 1;
    dp[4] = 2;
    dp[5] = 1;
    for (int i = 6; i <= n; i++)
    {
            dp[i] =dp[i - 2] + 1;
            dp[i] = min(dp[i], dp[i - 5] + 1);
 
 
    }
    if (dp[n] == 987654321)
        cout << -1;
    else
    cout << dp[n];

    return 0;
}